aviary/subsystems/propulsion/engine_model.py

```diff
@@ -135,22 +135,44 @@ def _preprocess_inputs(self):
         for (key, (val, units)) in options:
             # only perform vector check for variables related to engines and nacelles
             if key.startswith('aircraft:engine:') or key.startswith('aircraft:nacelle'):
+                # if val is an iterable...
                 if type(val) in (list, np.ndarray, tuple):
-                    if self.meta_data[key]['types'] not in (list, np.ndarray, tuple) and verbosity >= 1:
-                        warnings.warn(
-                            f'The value of {key} passed to EngineModel <{self.name}> is '
-                            f'type {type(val)}. Only the first entry in this iterable will '
-                            'be used.')
+                    # but meta_data says it is not supposed to be, use first item and
+                    # warn user
+                    if not isinstance(self.meta_data[key]['default_value'],
+                                      (list, np.ndarray, tuple)):
+
+                        # if val is multidimensional, raise error
+                        if isinstance(val[0], (list, np.ndarray, tuple)):
+                            raise UserWarning(f'Multidimensional {type(val)} was given '
```

Comment:
This is supposed to be an error - if you get a multidimensional array where only a scalar was expected I decided to hard stop since something went very wrong with how the user set up the input deck.

Reply:
so the raise should be an Error then?

Reply:
UserWarning actually inherits from Exception, so when you raise it becomes an error